pkg/compact/compact.go

```diff
@@ -1008,12 +1018,23 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 			uniqueSources[s] = struct{}{}
 		}
 
-		if err := block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir); err != nil {
+		// creating a child block span's context, identified by the block ID.
+		blockSpan, blockCtx := tracing.StartSpan(ctx, "block_operations", opentracing.Tags{"block ID": meta.ULID})
+		defer blockSpan.Finish()
+
+		tracing.DoInSpan(blockCtx, "compaction_block_download", func(ctx context.Context) {
+			err = block.Download(ctx, cg.logger, cg.bkt, meta.ULID, bdir)
+		}, opentracing.Tags{"block ID": meta.ULID})
```

Comment:
Let's use `block.id` for this tag. This seems like a naming convention for tracing tags.